File: tests/code/corpus/langs/swift/Shapes.swift
import Foundation

protocol Drawable {
    func draw()
}

class Shape: Drawable {
    func draw() {
    }

    func area() -> Double {
        return 0.0
    }
}

class Circle: Shape {
    private let radius: Double

    init(radius: Double) {
        self.radius = radius
    }

    override func area() -> Double {
        return Double.pi * radius * radius
    }
}

struct Point {
    var x: Double
}

func makeCircle(radius: Double) -> Circle {
    return Circle(radius: radius)
}

func totalArea(shapes: [Shape]) -> Double {
    return shapes.reduce(0.0) { $0 + $1.area() }
}

File: tests/code/corpus/langs/php/Shapes.php
<?php

namespace App\Geometry;

use App\Support\Formatter;

interface Drawable
{
    public function draw();
}

trait Loggable
{
    public function log()
    {
        return true;
    }
}

class Shape implements Drawable
{
    public function draw()
    {
    }

    public function area()
    {
        return 0.0;
    }
}

class Circle extends Shape
{
    private $radius;

    public function __construct($radius)
    {
        $this->radius = $radius;
    }

    public function area()
    {
        return M_PI * $this->radius * $this->radius;
    }
}

function makeCircle($radius)
{
    return new Circle($radius);
}

function totalArea($shapes)
{
    $total = 0.0;
    foreach ($shapes as $shape) {
        $total += $shape->area();
    }
    return $total;
}
